pyfr/integrators/base.py

```diff
@@ -64,18 +64,33 @@ def __init__(self, backend, rallocs, mesh, initsoln, cfg):
         # Abort computation
         self.abort = False
 
-    def _get_plugins(self):
+    def _get_plugins(self, initsoln):
         plugins = []
 
         for s in self.cfg.sections():
             if (m := re.match('soln-plugin-(.+?)(?:-(.+))?$', s)):
                 cfgsect, name, suffix = m.group(0), m.group(1), m.group(2)
 
+                data = {}
+                if initsoln:
+                    # get the plugin data stored in the solution, if any
+                    prefix = self.get_plugin_data_prefix(name, suffix)
+                    for f in initsoln:
+                        if re.match(prefix + '/', f):
+                            data[f.split('/')[-1]] = initsoln[f]
+
                 # Instantiate
-                plugins.append(get_plugin(name, self, cfgsect, suffix))
+                plugins.append(get_plugin(name, self, cfgsect, suffix, **data))
 
         return plugins
 
+    @staticmethod
+    def get_plugin_data_prefix(plugin):
+        prefix = f'plugins/{plugin.name}'
+        if plugin.suffix:
+            prefix += f'-{plugin.suffix}'
```

Comment:
Avoid using += on strings.  if/else two returns is probably easier.

Reply:
fixed in PR242